### PR TITLE
ARROW-17856: [CI][Archery] Add new Archery command to delete old branches and tags on crossbow repo

### PR DESCRIPTION
**Pull request by raulcd**
I have tested the script locally like:
```
$ archery crossbow delete-old-branches --days 600
Total number of references to delete: 342
```

Before running there were on the remote repository:
```
52,188 branches
50,866 tags
```
After running we have:
```
52,059 branches
50,768 tags
```

### CODE REVIEW

**Comment by raulcd** [comment]
@kszucs I've reused a snippet that you shared to do this but have moved it to a crossbow command. Let me know what do you think. I'll update the crossbow nightly job accordingly once this is merged.

File: dev/archery/archery/crossbow/cli.py [hunk 1]
```diff
@@ -551,3 +551,42 @@ def upload_artifacts(obj, tag, sha, patterns, method):
...
+@crossbow.command()
+@click.option('--dry-run/--execute', default=False,
+              help='Just display process, don\'t download anything')
+@click.option('--days', default=90,
+              help='Branches older than this amount of days will be deleted')
+@click.pass_obj
+def delete_old_branches(obj, dry_run, days):
+    """
+    Deletes branches on queue repository (crossbow) that are older than number
+    of days.
```
Comment (assignUser) [inline dev/archery/archery/crossbow/cli.py:565]: I would add a caveat here that deleting to many branches at once can trigger githubs hacking/fraud protection, though I do not know what the threshold is for that.

**Review by assignUser — APPROVED** [review]
Minor comment

File: dev/archery/archery/crossbow/cli.py (hunk 1 above)
Comment (raulcd) [reply dev/archery/archery/crossbow/cli.py:567]: Good idea, I've added a new CLI argument to limit the maximum number of branches to delete.

File: dev/archery/archery/crossbow/cli.py [hunk 2]
```diff
@@ -558,11 +558,14 @@ def upload_artifacts(obj, tag, sha, patterns, method):
...
               help='Branches older than this amount of days will be deleted')
+@click.option('--maximum', default=1000,
+              help='Maximum limit of branches to delete for a single run')
 @click.pass_obj
-def delete_old_branches(obj, dry_run, days):
+def delete_old_branches(obj, dry_run, days, maximum):
     """
     Deletes branches on queue repository (crossbow) that are older than number
     of days.
+    With a maximum number of branches to be deleted.
```
Comment (assignUser) [inline dev/archery/archery/crossbow/cli.py:568]: Should we add a note like "deleting to many branches at once can trigger the Github abuse protection"?

Comment (raulcd) [reply dev/archery/archery/crossbow/cli.py]: I've added a comment about why is required.

**Comment by raulcd** [comment]
@kszucs @kou this is ready for review. Thanks!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**